src/components/DatePicker/index.js

```diff
@@ -21,9 +21,9 @@ class Datepicker extends React.Component {
         /* We're using uncontrolled input otherwise it wont be possible to
         * raise change events with a date value - each change will produce a date
         * and make us reset the text input */
-        this.defaultValue = props.value
-            ? moment(props.value).format(CONST.DATE.MOMENT_FORMAT_STRING)
-            : '';
+        this.defaultValue = props.defaultValue
```

Comment:
Since choosing between defaultValue and value is appropriately chosen internally by `BaseTextInput`, We should pass down `defaultValue` to conform with the logic of `Form.js`.